suite-common/wallet-graph/src/graphReducer.ts

```diff
@@ -27,32 +28,36 @@ export type GraphRootState = {
     };
 };
 
-const updateSectionPoints = (
+const updateGraphPoints = (
     state: GraphState,
     payload: {
-        section: 'dashboard' | 'account';
-        points: LineGraphPoint[];
+        graphDisplayLocation: GraphDisplayLocation;
+        graphPoints: LineGraphPoint[];
     },
 ) => {
-    const { section, points } = payload;
-    /**
-     * react-native-graph library has problems with rendering path when there are some invalid values.
-     * Also graph is not showing (with props animated=true) when dates do not follow each other by milliseconds.
-     */
-    const validGraphPoints = points
-        .filter(point => !Number.isNaN(point.value))
-        .map((point, index) => ({
-            ...point,
-            date: new Date(index),
-        }));
-    state[section].points = validGraphPoints;
+    const { graphDisplayLocation, graphPoints } = payload;
+    state[graphDisplayLocation].points = getValidGraphPoints(graphPoints);
 };
 
-export const prepareGraphReducer = createReducerWithExtraDeps(graphInitialState, builder => {
-    builder.addCase(getGraphPointsForAccountsThunk.fulfilled, (state, action) => {
-        updateSectionPoints(state, action.payload);
-    });
+export const graphReducer = createReducer(graphInitialState, builder => {
+    builder
+        .addCase(getAllAccountsGraphPointsThunk.fulfilled, (state, action) => {
+            if (action.payload) {
+                updateGraphPoints(state, action.payload);
+            }
```

Comment:
Perhaps to avoid nesting and to be clearer with what the happy path is, similar things could be written this way?

```suggestion
            if (!action.payload) {
               return
            }
            
            updateGraphPoints(state, action.payload);
```

Reply:
I won't have it with the condition anyway. I added .rejected case instead of as the thunk can be rejected with "not found" and it is still a valid case. https://github.com/trezor/trezor-suite/commit/11e4de766fb57bfccbb6ea85dd9d8e4063d9703a#diff-51e61083ebf5590edfa4f0b1a034f88aad391bcfb9de034ae884e9d251d7d751R49